### PR TITLE
feat: normalization support for assortativity, and documentation updates

### PR DESCRIPTION
**Pull request by szhorvat**
Changes:

 - Update the docs of assortativity. This was done quickly, feel free to push to this PR and improve it.
 - Add a `normalized` parameter to support non-normalized assortativity.
 - Do not refer to vertex values as "types" for valued assortativity.

### CODE REVIEW

**Comment by szhorvat** [comment]
This is ready for review, but please do not merge. I might polish it a bit.

**Review by ntamas — APPROVED** [review]

**Comment by ntamas** [comment]
Seems okay to me! I won't merge now as you've requested, let me know when it's good to go.

**Review by ntamas — APPROVED** [review]

**Comment by ntamas** [comment]
Can you update the changelog pls after this is merged?

**Comment by szhorvat** [comment]
Yes, but this was supposed to be a squash-and-merge. Now we have a bunch of ugly messy commits with non-descriptive commit messages. Do you want to squash them after the fact or we leave them @ntamas ?

**Comment by vtraag** [comment]
Sorry, I was just looking through it at the same time as you apparently @ntamas . Only some minor nitpicks, all the rest looks fine to me!

File: src/misc/mixing.c
Comment (szhorvat) [reply src/misc/mixing.c:79]: Yes, we should. This was a mistake. I'll revert this.

**Comment by szhorvat** [comment]
Thanks @vtraag, I'll fix these tomorrow.

**Pull request merged**
